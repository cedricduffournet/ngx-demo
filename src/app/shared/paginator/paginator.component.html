
  <pagination
    class="mt-2"
    [totalItems]="totalItems"
    [maxSize]="20"
    [rotate]="false"
    [ngModel]="page"
    [itemsPerPage]="itemsPerPage"
    (pageChanged)="onPageChanged($event)"
    previousText="{{ 'PREVIOUS' | translate }}"
    nextText="{{ 'NEXT' | translate }}"
  ></pagination>


