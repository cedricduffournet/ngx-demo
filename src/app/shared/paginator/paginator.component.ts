import {
  Component,
  Input,
  ChangeDetectionStrategy,
  Output,
  EventEmitter
} from '@angular/core';

@Component({
  selector: 'app-paginator',
  templateUrl: './paginator.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class PaginatorComponent {
  @Input() page = 0;
  @Input() itemsPerPage = 0;
  @Input() totalItems = 0;
  @Output() pageChanged = new EventEmitter();

  onPageChanged(value: any) {
    this.pageChanged.emit(value.page);
  }
}
