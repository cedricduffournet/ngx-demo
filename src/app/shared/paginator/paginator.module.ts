import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';

import { PaginationModule } from 'ngx-bootstrap/pagination';
import { TranslateModule } from '@ngx-translate/core';
import { PaginatorComponent } from '@app/shared/paginator/paginator.component';

@NgModule({
  imports: [FormsModule, CommonModule, TranslateModule, PaginationModule],
  exports: [PaginatorComponent],
  declarations: [PaginatorComponent]
})
export class PaginatorModule {}
