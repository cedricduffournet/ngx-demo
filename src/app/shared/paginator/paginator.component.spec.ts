import { TestBed, ComponentFixture } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { FormsModule } from '@angular/forms';

import { PaginationModule } from 'ngx-bootstrap/pagination';
import { TranslateModule } from '@ngx-translate/core';

import { PaginatorComponent } from '@app/shared/paginator/paginator.component';

describe('PaginatorComponent', () => {
  let fixture: ComponentFixture<PaginatorComponent>;
  let component: PaginatorComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        TranslateModule.forRoot(),
        PaginationModule.forRoot(),
        FormsModule
      ],
      declarations: [PaginatorComponent]
    });

    fixture = TestBed.createComponent(PaginatorComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should emit a pageChanged when page change', () => {
    spyOn(component.pageChanged, 'emit');
    fixture.detectChanges();
    const pagination = fixture.debugElement.query(By.css('pagination'));
    pagination.triggerEventHandler('pageChanged', { page: 1 });
    expect(component.pageChanged.emit).toHaveBeenCalledWith(1);
  });

  it('should have 5 page', () => {
    component.itemsPerPage = 10;
    component.totalItems = 50;
    fixture.detectChanges();
    const paginationList = fixture.nativeElement.querySelectorAll(
      '.pagination-page'
    );
    expect(paginationList.length).toBe(5);
  });
});
