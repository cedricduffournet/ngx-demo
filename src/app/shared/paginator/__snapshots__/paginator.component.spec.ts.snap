// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatorComponent should be created 1`] = `
<app-paginator
  itemsPerPage="0"
  page="0"
  pageChanged={[Function EventEmitter]}
  totalItems="0"
>
  <pagination
    class="mt-2 ng-untouched ng-pristine ng-valid"
    ng-reflect-items-per-page="0"
    ng-reflect-max-size="20"
    ng-reflect-model="0"
    ng-reflect-next-text="NEXT"
    ng-reflect-previous-text="PREVIOUS"
    ng-reflect-rotate="false"
    ng-reflect-total-items="0"
  >
    <ul
      class="pagination mt-2"
      ng-reflect-klass="pagination"
      ng-reflect-ng-class="mt-2"
    >
      
      
      <li
        class="pagination-prev page-item disabled"
      >
        <a
          class="page-link"
          href=""
        >
          PREVIOUS
        </a>
      </li>
      
      <li
        class="pagination-page page-item active"
      >
        <a
          class="page-link"
          href=""
        >
          1
        </a>
      </li>
      
      <li
        class="pagination-next page-item disabled"
      >
        <a
          class="page-link"
          href=""
        >
          NEXT
        </a>
      </li>
      
    </ul>
  </pagination>
</app-paginator>
`;
